packages/headless/src/app/insight-engine/insight-engine.ts

```diff
@@ -0,0 +1,112 @@
+import {StateFromReducersMapObject} from '@reduxjs/toolkit';
+import {NoopPreprocessRequest} from '../../api/preprocess-request';
+import {InsightAPIClient} from '../../api/service/insight/insight-api-client';
+import {InsightAppState} from '../../state/insight-app-state';
+import {
+  buildEngine,
+  CoreEngine,
+  EngineOptions,
+  ExternalEngineOptions,
+} from '../engine';
+import {InsightThunkExtraArguments} from '../insight-thunk-extra-arguments';
+import {buildLogger} from '../logger';
+import {insightConfiguration} from '../reducers';
+import {buildThunkExtraArguments} from '../thunk-extra-arguments';
+import {
+  InsightEngineConfiguration,
+  insightEngineConfigurationSchema,
+} from './insight-engine-configuration';
+import {Logger} from 'pino';
+import {setInsightConfiguration} from '../../features/insight-configuration/insight-configuration-actions';
+
+export type {InsightEngineConfiguration};
+
+const insightEngineReducers = {
+  insightConfiguration,
+};
+type InsightEngineReducers = typeof insightEngineReducers;
+
+type InsightEngineState = StateFromReducersMapObject<InsightEngineReducers> &
+  Partial<InsightAppState>;
+
+export interface InsightEngine<State extends object = {}>
+  extends CoreEngine<State & InsightEngineState, InsightThunkExtraArguments> {}
+
+/**
+ * The insight engine options.
+ */
+export interface InsightEngineOptions
+  extends ExternalEngineOptions<InsightEngineState> {
+  /**
+   * The insight engine configuration options.
+   */
+  configuration: InsightEngineConfiguration;
+}
+
+/**
+ * Creates an insight engine instance.
+ *
+ * @param options - The insight engine options.
+ * @returns An insight engine instance.
+ */
+export function buildInsightEngine(
+  options: InsightEngineOptions
+): InsightEngine {
+  const logger = buildLogger(options.loggerOptions);
+  validateConfiguration(options.configuration, logger);
+
+  const insightAPIClient = createInsightAPIClient(
+    options.configuration,
+    logger
+  );
+
+  const thunkArguments: InsightThunkExtraArguments = {
+    ...buildThunkExtraArguments(options.configuration, logger),
+    apiClient: insightAPIClient,
+  };
+
+  const augmentedOptions: EngineOptions<InsightEngineReducers> = {
```

Comment:
The type of configuration mismatch is mainly searchhub or pipeline type things baked into the JWT and passed via options right?
In our case we're using a platform token instead of a search and the searchhub and pipeline values are configured and stored in our backend instead. So maybe i'm wrong but i don't think there's currently any overlap possible. It's a good thing to lookout for though.

Reply:
Thanks for the heads up regarding the `jwt-reducer`. However, I'm coming to the same conclusion as Nathan, and I think it does not apply to the Insight use case because the back-end gets to decide which search hub/pipline to use.